File: py/rt/ray.py
from vec3 import vec3

class ray:
	def __init__(self, o, d):
		self.o = o
		self.d = d

	def __str__(self):
		return "{" + str(self.o) + " + " + str(self.d) + " * t}"

	def origin(self):
		return self.o
		
	def direction(self):
		return self.d
		
	def pointAtParameter(self, t):
		return self.o + self.d*t
		
		
if __name__ == "__main__":
	x = ray(vec3(0,0,0), vec3(0,0,-1))
	print(x)	

File: cpp/OpenGL/22_Raytrace/Ray.cpp
#include "Ray.h"


Ray::Ray(const Ray& other) :
	o{other.o},
	d{other.d} 
{}

Ray::Ray(const Vec3& o, const Vec3& d) :
	o{o},
	d{d} 
{}

const Vec3 Ray::origin() const {
	return o;
}

const Vec3 Ray::direction() const {
	return d;
}

const Vec3 Ray::pointAtParameter(float t) const {
	return o + d*t;
}
